src/Compilers/Core/Portable/CommandLine/AnalyzerConfigSet.cs

```diff
@@ -146,6 +164,18 @@ public AnalyzerConfigOptionsResult GetOptionsForSourcePath(string sourcePath)
 
             var normalizedPath = PathUtilities.NormalizeWithForwardSlash(sourcePath);
 
+            // If we have a global config, add any sections that match the full path 
+            if (_globalConfig is object)
+            {
+                foreach (var section in _globalConfig.NamedSections)
```

Comment:
What about the global section? I thought the design was that everything in the global section matches, and then every named section that has an identical path

Reply:
Yeah, this is just building the lookup key. so the global section is always implicitly included (we don't need to always add it to match it in the cache)

---
In reply to: [425945443](https://github.com/dotnet/roslyn/pull/43889#discussion_r425945443) [](ancestors = 425945443)